deploymentframeworks/arm/output.ftl

```diff
@@ -106,7 +106,10 @@
             [@armOutput
                 name=name
                 type="string"
-                value="[resourceId(" + resourceIdParameters?join(", ") + ")]"
+                value=formatAzureResourceIdReference(
+                        resourceId=name
+                        resourceType=type
+                    )
```

Comment:
So in this case when creating an output it can lookup the ID of a resource which hasn't been deployed?

Reply:
Exactly. If it exists within the same template you would use it as in the example you linked, whereas if it was in another template or just happens to be previously deployed the function can be provided greater information. 

It's recreating this output from the network component of Epic 1:
```
"outputs": {
  "subnetId": {
    "type": "string",
    "value": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('vnet').name, 'default')]"
  }
}
```

Reply:
@roleyfoley I have rejigged this somewhat as I realised the need to implement both Azure ARM functions resourceId() (as you just reviewed) and reference() which returns a full resource not just the Id. Would you mind taking another look at the changes?